mscore/resourceManager.cpp

```diff
@@ -248,6 +262,142 @@ void ResourceManager::displayLanguages()
+            uninstall_button->setProperty("row", row);
+            uninstall_button->setEnabled(false);
+            connect(uninstall_button, SIGNAL(clicked()), this, SLOT(uninstallPluginPackage()));
+
+            pluginsTable->setIndexWidget(pluginsTable->model()->index(row, col++), install_button);
+            pluginsTable->setIndexWidget(pluginsTable->model()->index(row, col), uninstall_button);
+            refreshPluginButton(row);
+            }
+      categories->insertItems(0,all_categories);
+      }
+
+void ResourceManager::filterPluginList()
+      {
+      const QString category = categories->currentText();
+      const QString& search = lineEdit->text();
+      for (int i = 0; i < pluginsTable->rowCount(); i++) {
+            // locate the "Install" button
+            QPushButton* install = static_cast<QPushButton*>(pluginsTable->indexWidget(pluginsTable->model()->index(i, 3)));
+            const QString& plugin_name = install->property("name").toString();
+            const QStringList& categories = install->property("categories").toStringList();
+            if (plugin_name.contains(search, Qt::CaseInsensitive) && (category == "Any" || categories.contains(category)))
+                  pluginsTable->showRow(i);
+            else
+                  pluginsTable->hideRow(i);
+            }
+      }
+
+void ResourceManager::displayPlugins()
+      {
+      pluginTreeWidget->clear();
+      foreach(const QString &pkg, pluginDescriptionMap.keys()) {
+            auto& desc = pluginDescriptionMap[pkg];
+            auto* package_item = new QTreeWidgetItem(pluginTreeWidget);
+            package_item->setData(0, Qt::DisplayRole, desc.package_name);
+            package_item->setFlags(package_item->flags() | Qt::ItemIsEnabled);
+            package_item->setCheckState(0, Qt::Unchecked);
+            // add qmls
+            for (const auto &qml : desc.qml_paths) {
+                  QFileInfo f(qml);
+                  auto* qml_item = new QTreeWidgetItem(package_item);
+                  qml_item->setData(0, Qt::DisplayRole, f.fileName());
+                  qml_item->setFlags(qml_item->flags() | Qt::ItemIsEnabled);
+                  qml_item->setCheckState(0, Qt::Unchecked);
+                  }
+            }
+      for (int i = 0; i < mscore->getPluginManager()->pluginCount(); i++) {
+            auto* desc = mscore->getPluginManager()->getPluginDescription(i);
+            if (isPluginLocal(*desc)) {
+                  QTreeWidgetItem* item = new QTreeWidgetItem(pluginTreeWidget);
+                  item->setFlags(item->flags() | Qt::ItemIsEnabled);
+                  item->setCheckState(0, desc->load ? Qt::Checked : Qt::Unchecked);
+                  item->setData(0, Qt::DisplayRole, QFileInfo(desc->path).completeBaseName());
+                  }
+
+            }
+      }
+
+bool ResourceManager::isPluginLocal(PluginDescription& desc)
+      {
+      desc.path;
+      foreach(const auto& item, pluginDescriptionMap.values())
```

Comment:
likewise

Reply:
Resolved in 77701599820befd5e5381af2075d63e6b5b3dd85.